packages/sit-onyx/src/utils/plugin.ts

```diff
@@ -0,0 +1,19 @@
+import { injectI18n, provideI18n, type ProvideI18nOptions } from "@/i18n";
+import { type Plugin, watchEffect } from "vue";
+
+export type OnyxPluginOptions = {
+  i18n?: ProvideI18nOptions;
+};
+
+export default {
+  install: (app, options) => {
+    provideI18n(options.i18n, app);
+    const i18n = app.runWithContext(() => injectI18n());
+    watchEffect(() =>
+      globalThis.document.body.style.setProperty(
+        "--onyx-global-optional-text",
+        `"${i18n.t.value("optional")}"`,
+      ),
+    );
+  },
```

Comment:
```suggestion
  install: (app, options) => {
    provideI18n(options.i18n, app);
    const i18n = app.runWithContext(() => injectI18n());
    const rootElement = app._container as HTMLElement;

    watchEffect(() => {
      rootElement.style.setProperty("--onyx-global-optional-text", `"${i18n.t.value("optional")}"`);
    });
  },
```

You can access the root element with `app._container`

Reply:
Unfortunately not when the plugin is loaded, because the mount is happening afterwards. And afaik there are no lifecycle hooks on app level